Userland/Libraries/LibCore/System.cpp

```diff
@@ -330,7 +330,9 @@ ErrorOr<void> sigaction(int signal, struct sigaction const* action, struct sigac
     return {};
 }
 
-#if defined(AK_OS_BSD_GENERIC)
+#if defined(AK_OS_SOLARIS)
+ErrorOr<SIG_TYP> signal(int signal, SIG_TYP handler)
```

Comment:
How about you `#define SCM_CREDS sig_t` at the top, and let the code flow as in other platforms?

Reply:
This snipped has nothing to do with SCM_CREDS/credentials,it's about signal handling,and the type of the signal() response is called SIG_TYP on Solaris,but sig_t on other BSDs and signal_t on Linux.

Reply:
That's exactly what I am asking. IMHO, if you do that define at the top of the header, the rest of the header and CPP file will not be changed. It will create a smaller diff (and less noise in the main code).

Reply:
So BSD_GENERIC and Linux have their #ifs at that point and Solaris should be checked at the top?Sorry,but that doesn't make much sense and you'll have a hard time understanding that code later if you define something like that far from where it's used.